.github/workflows/changelog-verify.yml

```diff
@@ -0,0 +1,42 @@
+# This runs the pre-commit hook logic to check whether the relevent
+# CHANGELOG.md files have been updated for both the root directory
+# and/or modules edited if they have been changed.
+
+name: Validate Changelogs
+
+on:
+  pull_request:
+    types: [opened, reopened, edited, synchronize]
+    paths-ignore:
+      - ".github/**"
+      - ".githooks/**"
+      - "docs/**"
+      - "bin/**"
+
+jobs:
+  changedfiles:
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      all: ${{ steps.changes.outputs.all}}
+    steps:
+        # Make sure we have some code to diff.
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changes
+        # Set outputs using the command.
+        run: echo "all=$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | xargs)" >> $GITHUB_OUTPUT
+
+  validate:
+    runs-on: ubuntu-latest
+      # require the first job to have ran
+    needs: changedfiles
+      # only run there are changed files
+    if: ${{needs.changedfiles.outputs.all}}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Verify changelogs
+        run: bash ./.githooks/pre-commit ${{needs.changedfiles.outputs.all}}
```

Comment:
From my understanding: on the client side, the `exit 1` prevents the commit unless the flag `--no-verify` is used and this is bypassed; for the GitHub workflow the `exit 1` will fail the test but it can still be merged if you wish. To change this I think you would have to specify that this test completes successfully in the repository settings. If this is the desired outcome I could change the workflow to assign the `do not merge` label to prevent merging if the test fails. 

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule

Reply:
We have a `do-not-merge` workflow here: https://github.com/pokt-network/pocket/blob/main/.github/workflows/do-not-merge.yml

The idea is that if we apply the label, it prevents the PR from being merged to master.

The reason I ask is because our usual workflow is:
1. Make a bunch of commits locally
2. Push up for review
3. Get feedback
4. Repeat steps 1-3 a few times
5. Update changelog
6. Squash & Merge the github PR into main

I don't think the script will be used as a precommit, but will be very much more valuable as a workflow.

Can you confirm that I'm not missing anything?